### DISCUSSION TITLE
Filling the path without ending search. Is it possible ?

### DISCUSSION THREAD

**Discussion by iharkatkavets** [junegunn/fzf#3761] [Q&A] [answered]
Hello. 

I apply `fzf` next way:
I run a command and enter `**` and then `<TAB>`, for example:
```sh
neovim ** <TAB>
```

`fzf` successfully list me the files that matches the entered search string. But my path is too long, and I need to find first directory, then second, and then inner, and again and again. But I want to build the path folder by folder. And I want a magic shortcut that apply the path under cursor. Unfortunately I didn't find a shortcut to fill the path under cursor position, so I need to type the whole path, symbol by symbol.

Is there a shortcut/keymap to fill the path under cursor and continuously find the files/folders in the completed path?

Thank you

**Comment by iharkatkavets** [comment]
For example I selected 2nd item in the search list, but it's not final path, it can go deeper
<img width="498" alt="image" src="https://github.com/junegunn/fzf/assets/1083951/1206c44d-d808-48d8-9837-b42f43928bbb">

**Comment by LangLangBart** [comment, marked as answer]
> `fzf` successfully list me the files that matches the entered search string. But my path is too long, and I need to find first directory, then second, and then inner, and again and again.

Currently, `fzf` does not provide a way to traverse the file path as you described. Have you considered using plugins?

An exemplary tool is `zsh4humans`[^1]. In the `GIF` below, the filepath is constructed one hierarchy at a time until a file is selected.

<img src="https://github.com/junegunn/fzf/assets/92653266/f830abd7-b54b-4654-a524-8fff941f0c33" width="600">

[^1]: [zsh4humans/fn/z4h-fzf-complete at v5 · romkatv/zsh4humans · GitHub](https://github.com/romkatv/zsh4humans/blob/v5/fn/z4h-fzf-complete)
